src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,30 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+	
+	protected String value;
+	protected boolean isPrivate;
+	
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Phone // instanceof handles nulls
```

Comment:
Note that since equals method is pushed to the parent class, you should refer to the parent class for object comparisons. In this case, you should change Phone to Contact.